File: 6_catkin_ws/src/ssr_py_pkg/scripts/yao_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String

if __name__ == "__main__":
    rospy.init_node("yao_node")   #参数为这个节点的名称
    rospy.logwarn("yao_node节点初始化成功！")

    pub = rospy.Publisher("Adiran_Second_Topic",String,queue_size=10)

    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        rospy.loginfo("开始发送消息！")
        msg = String()
        msg.data = "Adiran_Second_msg"
        pub.publish(msg)
        rate.sleep()

File: 5_catkin_ws/src/sensor_pkg/src/yao_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "yao_node");

    //printf("Hello World!\n");
    //printf("你好 世界\n");

    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<std_msgs::String>("Adiran_Second_Topic",10);

    ros::Rate loop_rate(10);

    while(ros::ok())
    {
        printf("World Hello!\n");
        std_msgs::String msg;
        msg.data = "Adiran_Second_msg";
        pub.publish(msg);
        loop_rate.sleep();
    }
    
    return 0;
}
